core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -880,15 +880,12 @@ public EnvVars getEnvironment(TaskListener log) throws IOException, InterruptedE
         if (ws!=null)   // if this is done very early on in the build, workspace may not be decided yet. see HUDSON-3997
             env.put("WORKSPACE", ws.getRemote());
 
-        project.getScm().buildEnvVars(this,env);
+        project.getScm().buildEnvironment(this, env);
 
         if (buildEnvironments!=null)
             for (Environment e : buildEnvironments)
                 e.buildEnvVars(env);
 
-        for (EnvironmentContributingAction a : getActions(EnvironmentContributingAction.class))
-            a.buildEnvVars(this,env,getBuiltOn());
```

Comment:
Do not disagree, both ways are risky as I said in the original comment. I will test EnvInject tomorrow to check if it falls apart. Generally it requires more @jenkinsci/code-reviewers to weight-in.